File: app/src/main/java/com/discovr/discord/ui/main/MainEvent.java
package com.discovr.discord.ui.main;

public interface MainEvent {
    final class RollDice implements MainEvent {
        public RollDice() {
        }
    }

    final class SwipeLeft implements MainEvent {
        private final int position;

        public SwipeLeft(int position) {
            this.position = position;
        }

        public int getPosition() {
            return position;
        }
    }

    final class SwipeRight implements MainEvent {
        private final int position;

        public SwipeRight(int position) {
            this.position = position;
        }

        public int getPosition() {
            return position;
        }
    }

    final class LoadCards implements MainEvent {
    }
}

File: app/src/main/java/com/discovr/discord/ui/main/MainEvent.kt
package com.discovr.discord.ui.main

import android.view.MenuItem
import com.discovr.discord.model.Tag

interface MainEvent {
    class MenuEvent(val menuItems: Sequence<MenuItem>,
                    val areSets: HashMap<Tag, Boolean>,
                    val colorIds: HashMap<Tag, Int>) : MainEvent

    class FragmentStart : MainEvent

    class SwipeLeft : MainEvent

    class SwipeRight : MainEvent

    class DrinkClick(val item: MenuItem,
                     val isSet: Boolean,
                     val colorId: Int,
                     val tag: Tag) : MainEvent

    class HardcoreClick(val item: MenuItem,
                        val isSet: Boolean,
                        val colorId: Int,
                        val tag: Tag) : MainEvent
}

